### PR TITLE
mediawiki: add support for postgresql

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Mic92** [comment]
@ofborg test tests.mediawiki.mysql tests.mediawiki.postgresql

**Comment by Mic92** [comment]
@ofborg test mediawiki.mysql mediawiki.postgresql

File: nixos/modules/services/web-apps/mediawiki.nix [hunk 1]
```diff
@@ -87,7 +96,8 @@ let
 
       ## Database settings
       $wgDBtype = "${cfg.database.type}";
-      $wgDBserver = "${cfg.database.host}:${if cfg.database.socket != null then cfg.database.socket else toString cfg.database.port}";
+      $wgDBserver = "${dbAddr}";
+      $wgDBport = "${toString cfg.database.port}";
```
Comment (Mic92) [inline nixos/modules/services/web-apps/mediawiki.nix:100]: With postgresql this field is always mandatory and ignored for mysql: https://www.mediawiki.org/wiki/Manual:$wgDBport

File: nixos/modules/services/web-apps/mediawiki.nix [hunk 2]
```diff
@@ -431,8 +455,10 @@ in
     systemd.services.mediawiki-init = {
       wantedBy = [ "multi-user.target" ];
       before = [ "phpfpm-mediawiki.service" ];
-      after = optional cfg.database.createLocally "mysql.service";
+      after = optional (cfg.database.type == "mysql" && cfg.database.createLocally) "mysql.service"
+              ++ optional (cfg.database.type == "postgres" && cfg.database.createLocally) "postgresql.service";
       script = ''
+        set -x
```
Comment (mweinelt) [inline nixos/modules/services/web-apps/mediawiki.nix:461]: Do you want to keep that?

Comment (Mic92) [reply nixos/modules/services/web-apps/mediawiki.nix:461]: Nope.

**Comment by Mic92** [comment]
@ofborg test mediawiki.mysql mediawiki.postgresql

**Review by mweinelt — APPROVED** [review]
Changes look good to me. Thanks!

File: nixos/modules/services/web-apps/mediawiki.nix (hunk 2 above)
Comment (mweinelt) [reply nixos/modules/services/web-apps/mediawiki.nix:461]: Forgot to push?

Comment (Mic92) [reply nixos/modules/services/web-apps/mediawiki.nix]: yes.

**Pull request merged**